spock-specs/src/test/groovy/org/spockframework/geantyref/GenericTypeReflectorUtilSpec.groovy

```diff
@@ -0,0 +1,135 @@
+//file:noinspection GrUnnecessaryPublicModifier
+//file:noinspection unused
+package org.spockframework.geantyref
+
+import groovy.transform.Canonical
+import groovy.transform.ToString
+import io.leangen.geantyref.TypeToken
+import org.spockframework.runtime.model.NodeInfo
+import spock.lang.Specification
+
+import java.lang.annotation.Annotation
+import java.lang.reflect.AnnotatedElement
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+import java.lang.reflect.Type
+import java.util.function.Function
+
+class GenericTypeReflectorUtilSpec extends Specification {
+  private static final Type FUNC_STRING_INT_TYPE = new TypeToken<Function<String, Integer>>() {}.type
+  private static final Type LIST_STRING_TYPE = new TypeToken<List<String>>() {}.type
+  private static final Type LIST_LIST_STRING_TYPE = new TypeToken<List<List<String>>>() {}.type
+
+  def "erase"(Type inputType, Class<?> expectedClass) {
+    expect:
+    GenericTypeReflectorUtil.erase(inputType) == expectedClass
+
+    where:
+    inputType             | expectedClass
+    Object                | Object
+    StringBuilder         | StringBuilder
+    FUNC_STRING_INT_TYPE  | Function
+    LIST_STRING_TYPE      | List
+    LIST_LIST_STRING_TYPE | List
+  }
+
+  def "getTypeName"(Type inputType, String expectedName) {
+    expect:
+    GenericTypeReflectorUtil.getTypeName(inputType) == expectedName
+
+    where:
+    inputType             | expectedName
+    Object                | "java.lang.Object"
+    StringBuilder         | "java.lang.StringBuilder"
+    FUNC_STRING_INT_TYPE  | "java.util.function.Function<java.lang.String, java.lang.Integer>"
+    LIST_STRING_TYPE      | "java.util.List<java.lang.String>"
+    LIST_LIST_STRING_TYPE | "java.util.List<java.util.List<java.lang.String>>"
+  }
+
+  def "getParameterTypes"(TestMethod testMethod, List<Type> expectedTypes) {
+    expect:
+    GenericTypeReflectorUtil.getParameterTypes(testMethod.method, testMethod.declaringClass).toList() == expectedTypes
+
+    where:
+    testMethod                               | expectedTypes
+    method(Object, "equals")                 | [Object]
+    method(List, "equals")                   | [Object]
+    method(NodeInfo, "getAnnotationsByType") | [Class]
```

Comment:
What about the more complex methods below? Also add an example with multiple parameters.

Reply:
I have added more methods to that.